vars/runATH.groovy

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env groovy
+
+/**
+ * Wrapper for running the ATH
+ */
+
+def call(Map params = [:]) {
+    def String athUrl = params.get('athUrl', 'https://github.com/jenkinsci/acceptance-test-harness.git')
+    def String athRevision = params.get('athRevision', 'master')
+    def String metadataFile = params.get('metadataFile', 'essentials.yml')
+    def String jenkins = params.get('jenkins', 'latest')
+
+    def mirror = "http://mirrors.jenkins.io/"
+    def defaultCategory = "org.jenkinsci.test.acceptance.junit.SmokeTest"
+    def jenkinsURl = jenkins
+    def metadata
+    def athContainerImage
+    def isLocalATH
+    def isVersionNumber
+
+    def athSourcesFolder = "athSources"
+
+    def supportedBrowsers = ["firefox"]
+
+    def skipExecution = false
+
+
+    ensureInNode(env, env.RUN_ATH_SOURCES_AND_VALIDATION_NODE ?: "linux", {
+        List<String> env = [
+                "JAVA_HOME=${tool 'jdk8'}",
+                'PATH+JAVA=${JAVA_HOME}/bin',
+                "PATH+MAVEN=${tool 'mvn'}/bin"
+
+        ]
+
+        if (!fileExists(metadataFile)) {
+            echo "Skipping ATH execution because the metadata file does not exist. Current value is ${metadataFile}."
+            skipExecution = true
+            return
+        }
+
+        stage("Getting ATH sources and Jenkins war") {
+            // Start validation
+            metadata = readYaml(file: metadataFile).ath
+            if (metadata == null) {
+                error "The provided metadata file seems invalid as it does not contain a valid ath section"
```

Comment:
Is it the same if it doesn't have a metadata section, if it's not valid or if it's empty? I think we should treat this as separate cases. For example if empty, maybe we want to suggest a default. If not valid we want to bring that to the developer's attention and if not existing the error message would be different as well.

Reply:
From my PoV a metadata file should contain an ath section to be considered valid.
 ATH section itself is validated in next lines of code